### PR TITLE
[SPARK-20899][PySpark] PySpark supports stringIndexerOrderType in RFormula

### PR DESCRIPTION
**Pull request by actuaryzhang**
## What changes were proposed in this pull request?

PySpark supports stringIndexerOrderType in RFormula as in #17967. 

## How was this patch tested?
docstring test

### CODE REVIEW

**Comment by actuaryzhang** [comment]
@felixcheung @yanboliang @viirya

File: python/pyspark/ml/feature.py [hunk 1]
```diff
@@ -3043,26 +3055,35 @@ class RFormula(JavaEstimator, HasFeaturesCol, HasLabelCol, JavaMLReadable, JavaM
                             "Force to index label whether it is numeric or string",
                             typeConverter=TypeConverters.toBoolean)
 
+    stringIndexerOrderType = Param(Params._dummy(), "stringIndexerOrderType",
+                                   "How to order categories of a string FEATURE column used by " +
```
Comment (felixcheung) [inline python/pyspark/ml/feature.py]: FEATURE capitalize is common here?

Comment (actuaryzhang) [inline python/pyspark/ml/feature.py]: Changed it to lower case now.

File: python/pyspark/ml/feature.py [hunk 2]
```diff
@@ -3032,6 +3032,18 @@ class RFormula(JavaEstimator, HasFeaturesCol, HasLabelCol, JavaMLReadable, JavaM
...
+    'alphabetDesc'
+    >>> rf.fit(df).transform(df).show()
+    +---+---+---+---------+-----+
+    |  y|  x|  s| features|label|
+    +---+---+---+---------+-----+
+    |1.0|1.0|  a|[1.0,0.0]|  1.0|
+    |0.0|2.0|  b|[2.0,1.0]|  0.0|
+    |0.0|0.0|  a|(2,[],[])|  0.0|
+    +---+---+---+---------+-----+
+    ...
```
Comment (yanboliang) [inline python/pyspark/ml/feature.py]: Could you move the newly added test to ```tests.py```? We keep the basic doc tests here both for test and example, other tests should be placed at ```tests.py```. Thanks.

**Review by yanboliang — COMMENTED** [review]
One minor comment, otherwise LGTM. Thanks!

**Comment by actuaryzhang** [comment]
@yanboliang I have moved the tests to the test file. Please let me know if there is anything else needed. Thanks.

**Comment by viirya** [comment]
LGTM

File: python/pyspark/ml/tests.py [hunk 3]
```diff
@@ -538,6 +538,19 @@ def test_rformula_force_index_label(self):
...
+            (1.0, 1.0, "a"),
+            (0.0, 2.0, "b"),
+            (1.0, 0.0, "a")], ["y", "x", "s"])
+        rf = RFormula(formula="y ~ x + s", stringIndexerOrderType="alphabetDesc")
+        self.assertEqual(rf.getStringIndexerOrderType(), 'alphabetDesc')
+        transformedDF = rf.fit(df).transform(df)
+        observed = transformedDF.select("features").collect()
+        expected = [[1.0, 0.0], [2.0, 1.0], [0.0, 0.0]]
+        for i in range(0, len(expected)):
+            self.assertTrue((observed[i]["features"].toArray() == expected[i]).all())
```
Comment (yanboliang) [inline python/pyspark/ml/tests.py]: Minor: Usually we're more prefer to use ```self.assertTrue(all(observed[i]["features"].toArray() == expected[i]))```.

**Review by yanboliang — COMMENTED** [review]
One very minor comment, thanks!

**Comment by yanboliang** [comment]
Merged into master, thanks for all.

**Pull request closed without merging**